trees_emit.h

```diff
@@ -53,7 +53,9 @@ extern Z_INTERNAL const int base_dist[D_CODES];
         bi_buf = val;\
         bi_valid = len;\
     } else {\
-        bi_buf |= val << bi_valid;\
+        if (bi_valid < BIT_BUF_SIZE) {\
```

Comment:
I think the `else` should be `else if (bi_valid < BIT_BUF_SIZE) {`

Reply:
Fails on line 49 with that change...

Reply:
For some reason `bi_valid` ends up being more than `BIT_BUF_SIZE`, but I can't trace why or where it happens when using static analyzer...

Reply:
If this is an impossible state, maybe make sense to add an ``assert``?